File: 13-RomanToInteger/13-RomanToInteger.c
// Last updated: 5/8/2026, 4:23:50 PM
int romanToInt(char* s) 
{
    char roman[] = {'I','V','X','L','C','D','M'};
    int value[] = {1,5,10,50,100,500,1000};

    int ans = 0;
    int length = strlen(s);
    for(int i=0; i<length;i++)
    {
        int current, next = 0;

        for(int j=0; j<7; j++)
        {
            if(s[i] == roman[j])
                current = value[j];
            
            if(i+1 < length && s[i+1] == roman[j])
                next = value[j];
        }

        if(current < next)
            ans -= current;
        else
            ans += current;
    }
    return ans;
}

File: 13-RomanToInteger/13-RomanToInteger.py
# Last updated: 5/8/2026, 4:23:43 PM
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {
            "I":1, 
            "V":5, 
            "X":10, 
            "L":50, 
            "C":100, 
            "D":500, 
            "M":1000
        }

        total_sum = 0
        for i in range(len(s)):
            current_val = roman[s[i]]
            
            if i+1 < len(s):
                next_val = roman[s[i+1]]
                if current_val < next_val:
                    total_sum -= current_val
                else:
                    total_sum += current_val
            else:
                total_sum += current_val
        
        return total_sum